### PR TITLE
5081 fix missing records azuredevops

### LINKED ISSUE #5081: [Bug][Azure DevOps] Missing records in cicd_tasks
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

**Version: v0.17.0-beta3**

## Azure DevOps Setup

I have two pipeline runs in Azure DevOps, each with two jobs, "job_1" and "job_2".
Screen Shot 2023-05-03 at 6 50 34 PM
Screen Shot 2023-05-03 at 6 50 21 PM

## DevLake Collection Result

I got 2 cicd_pipelines, which was as expected. But I only got cicd_tasks for the second cicd_pipeline, see the pipeline_id column of the cicd_tasks I got. Also, all of their environment was set to "TESTING", which was unexpected.

I also didn't expect to get the "Finalize Build" job, but since it's also displayed as a job in the Azure DevOps pipeline page, so I think it's ok.

Screen Shot 2023-05-03 at 6 49 17 PM
Screen Shot 2023-05-03 at 6 50 03 PM

### What do you expect to happen

See above

### How to reproduce

See above

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by CamilleTeruel**
### Summary
Implements migration of tool tables.
Fix a non-unique primary key in azuredevops table

### Does this close any open issues?
Closes #5081

### CODE REVIEW

File: backend/python/README.md [hunk 1]
```diff
@@ -232,6 +232,25 @@ Do not forget `table=True`, otherwise no table will be created in the database.
...
+```python
+class User(ToolModel, table=True):
+    id: str = Field(primary_key=True)
+    name: str
+    email: str
+    age: int
+
+    @classmethod
+    def migrate(cls, session):
+        session.execute(f"ALTER TABLE {cls.__tablename__} ADD COLUMN age INT")
```
Comment (hezyin) [inline backend/python/README.md:251]: Hi @CamilleTeruel, some clarifying questions on how tool model migrations work:

1. When are these migrations run? Are they triggered from the Golang side?
2. Our Golang side keeps track of migrations by date so each time only new migrations are run. How do we avoid re-running migrations that have already been run? Would the `migrate` function throw an error if it's run on a later version of DevLake?

Comment (CamilleTeruel) [reply backend/python/README.md:251]: > Hi @CamilleTeruel, some clarifying questions on how tool model migrations work:
> 
>     1. When are these migrations run? Are they triggered from the Golang side?

Yes, after loading a remote plugin, go-side calls run-migration.
 
>     2. Our Golang side keeps track of migrations by date so each time only new migrations are run. How do we avoid re-running migrations that have already been run? 

Right now we don't. This is just minimal support for migration to support this change. We must add versioning to migration functions.

> Would the `migrate` function throw an error if it's run on a later version of DevLake?

A migrate function can indeed throw an error if it has already run once, so we should be careful with migration until we have versionning support. But the `Job.migrate` function in particular is fine as it executes idempotent SQL statements.

Comment (hezyin) [reply backend/python/README.md:251]: Hi @CamilleTeruel, thanks for the clarification. I'm merging this PR now. Could you please create a new GitHub issue for adding versioning to migration functions? Please also cherry-pick this fix back to release-v0.17 branch.

**Review by hezyin — APPROVED** [review]
LGTM

**Comment by CamilleTeruel** [comment]
@keon94  This PR is missing from v0.17-beta6

**Pull request merged**